src/universal/redux/makeReducer.js

```diff
@@ -6,15 +6,41 @@ import authToken from './authDuck';
 import {reducer as storageReducer} from 'redux-storage-whitelist-fn';
 import storageMerger from 'universal/redux/storageMerger';
 
-const currentReducers = {
+const initialReducers = {
   authToken,
   cashay: cashayReducer,
   form: formReducer,
   notifications
 };
+const currentReducers = {...initialReducers};
 
 export default (newReducers) => {
+  /**
+   * Add the initial state of the new reducers to the initialReducers object. If
+   * we later revert the application to the initial state -- say to logout --
+   * we'll need these dynamically loaded initial states present so
+   * mapStateToProps can run one last time before the component unmounts:
+   */
+  Object.assign(initialReducers, newReducers);
```

Comment:
Still gotta dig into this. There's got to be an eloquent way to unsubscribe from all of those listeners before calling `dispatch`. Will PR tomorrow